ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d.hpp

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/core.hpp"
+#include "ttnn/types.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/run_operation.hpp"
+#include "ttnn/tensor/host_buffer/functions.hpp"
+
+#include "device/max_pool2d_device_op.hpp"
+
+
+namespace ttnn {
+namespace operations::pool {
+
+struct MaxPoolNewOp {
+    template<typename T>
+    static Tensor operator()(uint8_t queue_id, const Tensor& input_tensor, uint32_t batch_size, uint32_t input_h, uint32_t input_w, uint32_t channels, std::array<uint32_t, 2> kernel_size, std::array<uint32_t, 2> stride, std::array<uint32_t, 2> padding, std::array<uint32_t, 2> dilation, T* device);
```

Comment:
I would love to see these args compressed into a struct. 
You will have to add Struct bindings for it to work too. That will keep options list sane
Wdyt?

Reply:
I would keep them as is because:
1. it matches the pytorch maxpool args list
2. we will eventually get rid of `batch_size`, `input_h`, `input_w` and `channels` args once the new tensor infra is ready.